src/gluonts/transform/sampler.py

```diff
@@ -45,6 +45,26 @@ def __call__(self, ts: np.ndarray) -> np.ndarray:
         raise NotImplementedError()
 
 
+class NInstanceSampler(InstanceSampler):
```

Comment:
I'm tempted to say that `NumInstanceSampler` would be a better name: just like `ExpectedNumInstanceSampler` samples a certain number of instances in expectations, this samples a certain number of instances (full stop).

Let's see also what others think about the naming maybe

Reply:
Let's say: if all these samplers had a more "assertive" naming, I would just call this `SampleN` (and then `SampleOnAverageN`)

Reply:
Makes sense. Updated.